File: implementation/scipamato/scipamato-core-persistence-jooq/src/main/java/ch/difty/scipamato/core/persistence/user/UserFilterConditionMapper.java
package ch.difty.scipamato.core.persistence.user;

import static ch.difty.scipamato.core.db.tables.ScipamatoUser.SCIPAMATO_USER;

import java.util.List;

import org.jooq.Condition;

import ch.difty.scipamato.common.persistence.AbstractFilterConditionMapper;
import ch.difty.scipamato.common.persistence.FilterConditionMapper;
import ch.difty.scipamato.core.entity.search.UserFilter;

/**
 * Mapper turning the provider {@link UserFilter} into a jOOQ {@link Condition}.
 *
 * @author u.joss
 */
@FilterConditionMapper
public class UserFilterConditionMapper extends AbstractFilterConditionMapper<UserFilter> {

    @Override
    public void map(final UserFilter filter, final List<Condition> conditions) {
        if (filter.getNameMask() != null) {
            final String likeExpression = "%" + filter.getNameMask() + "%";
            conditions.add(SCIPAMATO_USER.USER_NAME.likeIgnoreCase(likeExpression)
                .or(SCIPAMATO_USER.FIRST_NAME.likeIgnoreCase(likeExpression))
                .or(SCIPAMATO_USER.LAST_NAME.likeIgnoreCase(likeExpression)));
        }
        if (filter.getEmailMask() != null) {
            final String likeExpression = "%" + filter.getEmailMask() + "%";
            conditions.add(SCIPAMATO_USER.EMAIL.likeIgnoreCase(likeExpression));
        }
        if (filter.getEnabled() != null) {
            final boolean expression = filter.getEnabled();
            conditions.add(SCIPAMATO_USER.ENABLED.eq(expression));
        }
    }

}

File: core/core-persistence-jooq/src/main/java/ch/difty/scipamato/core/persistence/user/UserFilterConditionMapper.kt
package ch.difty.scipamato.core.persistence.user

import ch.difty.scipamato.common.persistence.AbstractFilterConditionMapper
import ch.difty.scipamato.common.persistence.FilterConditionMapper
import ch.difty.scipamato.core.db.tables.ScipamatoUser
import ch.difty.scipamato.core.entity.search.UserFilter
import org.jooq.Condition

/**
 * Mapper turning the provider [UserFilter] into a jOOQ [Condition].
 */
@FilterConditionMapper
class UserFilterConditionMapper : AbstractFilterConditionMapper<UserFilter>() {

    override fun internalMap(filter: UserFilter): List<Condition> {
        val conditions = mutableListOf<Condition>()
        filter.nameMask?.let {
            val likeExpression = "%$it%"
            conditions.add(
                ScipamatoUser.SCIPAMATO_USER.USER_NAME.likeIgnoreCase(likeExpression)
                    .or(ScipamatoUser.SCIPAMATO_USER.FIRST_NAME.likeIgnoreCase(likeExpression))
                    .or(ScipamatoUser.SCIPAMATO_USER.LAST_NAME.likeIgnoreCase(likeExpression))
            )
        }
        filter.emailMask?.let {
            conditions.add(ScipamatoUser.SCIPAMATO_USER.EMAIL.likeIgnoreCase("%$it%"))
        }
        filter.enabled?.let {
            conditions.add(ScipamatoUser.SCIPAMATO_USER.ENABLED.eq(it))
        }
        return conditions
    }
}
